src/main/java/org/jabref/logic/importer/SearchBasedParserFetcher.java

```diff
@@ -65,9 +62,9 @@ default URL getURLForQuery(ComplexSearchQuery query) throws URISyntaxException,
     /**
      * Constructs a URL based on the query.
      *
-     * @param query the search query
+     * @param transformedQuery the search query
      */
-    URL getURLForQuery(String query) throws URISyntaxException, MalformedURLException, FetcherException;
+    URL getURLForQuery(String transformedQuery, AbstractQueryTransformer transformer) throws URISyntaxException, MalformedURLException, FetcherException;
```

Comment:
Can you please explain why the `getURL*` methods need to have the transformer as a second argument? It seems that each fetcher has it's own transformer, so why not use a simple class variable?

Reply:
The main reason for this is that some transformers require state that is used during query construction (See IEEE) and/or postprocessing (See ArXiv).
I now use a class variable and a reset method. :)